internal/bgp/frr/config.go

```diff
@@ -33,20 +33,53 @@ hostname {{.Hostname}}
 ip nht resolve-via-default
 ipv6 nht resolve-via-default
 
+{{- range $community, $prefixes := .PrefixesV4ForCommunity }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$community}}-v4prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $community, $prefixes := .PrefixesV6ForCommunity }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$community}}-v6prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $localPref, $prefixes := .PrefixesV4ForLocalPref }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$localPref}}-v4localpref-prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+{{- range $localPref, $prefixes := .PrefixesV6ForLocalPref }}
+{{- range $p := $prefixes.Elements }}
+ip prefix-list {{$localPref}}-v6localpref-prefixes permit {{$p}}
+{{- end}}
+{{- end}}
+
 {{- range .Routers }}
 {{- range $n := .Neighbors }}
-{{- range $a := .Advertisements }}
-{{- if or (gt (len $a.Communities) 0) (ne $a.LocalPref 0) }}
-route-map {{$n.Addr}}-out permit 10
-{{- range $c := $a.Communities }}
-  set community {{$c}} additive
+route-map {{$n.Addr}}-in deny 20
+{{/* NOTE: it's possible to have global routes only because all the neighbors
+receive the same advertisements. Once that changes, we'll need prefix lists per neighbor */}}
+{{- range $localPref, $prefixes := $.PrefixesV4ForLocalPref }}
+route-map {{$n.Addr}}-out permit {{counter $n.Addr}}
+  match ip address prefix-list {{$localPref}}-v4localpref-prefixes
+  set local-preference {{$localPref}}
 {{- end }}
-{{- if ne $a.LocalPref 0 }}
-  set local-preference {{$a.LocalPref}}
+{{- range $localPref, $prefixes := $.PrefixesV6ForLocalPref }}
+route-map {{$n.Addr}}-out permit {{counter $n.Addr}}
+  match ipv6 address prefix-list {{$localPref}}-v6localpref-prefixes
+  set local-preference {{$localPref}}
 {{- end }}
+{{- range $community, $prefixes := $.PrefixesV4ForCommunity }}
+route-map {{$n.Addr}}-out permit {{ counter $n.Addr }}
+  match ip address prefix-list {{$community}}-v4prefixes
+  set community {{$community}} additive
 {{- end }}
+{{- range $community, $prefixes := $.PrefixesV6ForCommunity }}
+route-map {{$n.Addr}}-out permit {{ counter $n.Addr }}
+  match ipv6 address prefix-list {{$community}}-v6prefixes
+  set community {{$community}} additive
 {{- end }}
-route-map {{$n.Addr}}-in deny 20
+route-map {{$n.Addr}}-out permit {{ counter $n.Addr }}
 {{- end }}
 {{- end }}
 
```

Comment:
I knew from day 1 this template will hard to maintain and indeed its getting there now and we still have more config to comes,  I proposed breaking this to sub templates do u think its time to refactor this I have some code for that but its way out of date should we consider this for this release or wait to the next release ?

Reply:
I agree, I am fine with both. We have enough coverage now that should make us confident enough in doing the refactor.
Maybe I'd try to reach a stable point, cut a release and then refactor.